zerologr.go

```diff
@@ -131,7 +142,39 @@ func (ls LogSink) WithCallDepth(depth int) logr.LogSink {
 	return &ls
 }
 
+// WithTransformer returns a new LogSink that apply map/filter on key/value pairs.
+func (ls LogSink) WithTransformer(tf TransformerFunc) logr.LogSink {
+	ls.tf = tf
+	return &ls
+}
+
 // GetUnderlying returns the zerolog.Logger underneath this logSink.
 func (ls *LogSink) GetUnderlying() *zerolog.Logger {
 	return ls.l
 }
+
+func WithMarshaler(l Logger) Logger {
+	return WithTransformer(l, Transformer)
+}
+
+func WithTransformer(l Logger, tf TransformerFunc) Logger {
```

Comment:
I'm not keen on this particular pattern - it's going to require multiple calls to set up.

What about an options struct with a field to "support standard transformations"?

Or what about not making it an option?  How much does this hurt critical benchmarks?  I considered it important enough to default to "on" for funcr.